app/controllers/api/v2/personal_access_tokens_controller.rb

```diff
@@ -66,6 +66,19 @@ def parent_permission(child_perm)
           super
         end
       end
+
+      def parent_resource_details
+        # return nil skips the parent check, which is desired behaviour for current user
+        super unless editing_self?
+      end
+
+      def resource_scope(*args)
+        if editing_self?
+          parent_scope.where(user: @user).readonly(false)
```

Comment:
It might be useless to call parent scope now, isn't it? it will be parent.all, because of the above change.
We might just use `resource_class.where(user: @user)` here and drop the above change to the `parent_scope`

Reply:
Good idea! I tested and it works as expected.